app/views/teachers/JoinLicensesByCode.vue

```diff
@@ -0,0 +1,59 @@
+<script>
+import { mapActions } from 'vuex'
+export default {
+  data () {
+    return {
+      state: 'loading',
+      result: {}
+    }
+  },
+  mounted () {
+    if (this.$route.query.codes) {
+      this.post()
+    } else {
+      application.router.navigate('/teachers/licenses', { trigger: true })
```

Comment:
instead of this, just showing error message on this page would be better experience.